### PR TITLE
[PF-1289] Stop using pet SA key in integration tests

### PR DESCRIPTION
**Pull request by melissachang**
Also:

- Delete all usages of pet SA key in this repo. Eg stop storing keys in `.terra/pet-keys/`. When this repo was originally created, a human running cli on a local machine would authenticate as pet (using key file). Because SA key files are a security risk, humans now authenticate as themselves.
- In `Dockerfile` bump nextflow version because [`google.lifeSciences.keepAliveOnFailure`](https://www.nextflow.io/docs/latest/google.html) requires `21.06.0-edge` or later. `keepAliveOnFailure` doesn't work for me, but at least it's not because of version.

### CODE REVIEW

File: src/main/java/bio/terra/cli/app/utils/AppDefaultCredentialUtils.java [hunk 1]
```diff
@@ -19,19 +19,7 @@
 
   /** Return the absolute path to file backing the current application default credentials. */
   public static Optional<Path> getADCBackingFile() {
```
Comment (jaycarlton) [inline src/main/java/bio/terra/cli/app/utils/AppDefaultCredentialUtils.java:21]: nit: probably rename this `getAdcBackingFile()` to match convention.

File: src/main/java/bio/terra/cli/businessobject/User.java [hunk 2]
```diff
@@ -338,21 +261,21 @@ private void fetchUserInfo() {
...
         // otherwise, just load the pet SA
-        fetchPetSaCredentials();
+        fetchPetSaEmail();
       }
     } catch (Exception ex) {
-      logger.error("Error loading workspace or pet SA credentials during login", ex);
+      logger.error("Error loading workspace or pet SA email during login", ex);
       UserIO.getErr()
           .println(
               "Error loading workspace information for the logged in user (workspace id: "
```
Comment (jaycarlton) [inline src/main/java/bio/terra/cli/businessobject/User.java:281]: As a user, I'm not sure what I'd do with this message. Maybe indicate that the login continued despite the error?

File: src/test/java/harness/TestUser.java [hunk 3]
```diff
@@ -112,38 +112,40 @@ private void writeTerraOAuthCredentialFile(GoogleCredentials googleCredentials)
...
+   *
+   * <p>One thing is needed for `gsutil` auth:
+   *
+   * <ol>
+   *   <li>.boto
+   * </ol>
+   *
+   * <p>Note: Authentication with `gcloud alpha storage` is simpler. You only need to set
+   * CLOUDSDK_AUTH_ACCESS_TOKEN; you don't need .boto. But the rest of this repo uses `gsutil`, so
+   * set up for `gsutil` here. >>>>>>> 2f8efe2 ([PF-1289] Stop using pet SA key in integration
```
Comment (jaycarlton) [inline src/test/java/harness/TestUser.java:147]: is this part of a conflict indicator?

**Review by jaycarlton — APPROVED** [review]

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -216,29 +219,31 @@ The CLI uses the test users defined in test config (eg `testconfig/broad.json`).
...
-script must be an admin user (i.e. a member of the `fc-admins` Google group in the SAM Gsuite). The script
-invites all the test users if they do not already exist in SAM, and this requires admin permissions.
+If the current server requires users to be invited before they can register, then the user who runs this script must be
+an admin user (i.e. a member of the `fc-admins` Google group in the SAM Gsuite). The script invites all the test users
+if they do not already exist in SAM, and this requires admin permissions.
 
 For each test user, store refresh token in vault:
-- Run `gcloud auth login` for test user. For password, see Test Horde spreadsheet in Broad Google Drive.
+
+- Run `gcloud auth login` for test user. For password, see Test Horde spreadsheet in Broad Google Drive. =======
```
Comment (zloery) [inline CONTRIBUTING.md:237]: leftover `=======` from conflict

**Review by zloery — APPROVED** [review]

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -380,29 +385,26 @@ To add a new supported tool:
...
+      terra (e.g. `terra dsub -version`).
    6. When you run e.g. `terra dsub -version`, the CLI:
-      - Launches a Docker container
-      - Runs the `terra_init.sh` script in the `docker/scripts` directory, which activates the user’s
-      pet service account and sets the workspace project
-      - Runs the `dsub` command
-   7. You can pass environment variables through to the Docker container by populating a `Map` and
-   passing it to the `DockerAppsRunner.runToolCommand` method. Two environment variables are always
-   passed:
+       - Launches a Docker container
```
[the remark this answers is not in the document]
Comment (melissachang) [reply CONTRIBUTING.md]: Updated 6 and 7

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -380,29 +385,26 @@ To add a new supported tool:
...
+       - Runs the `terra_init.sh` script in the `docker/scripts` directory, which sets the workspace project
+       - Runs the `dsub` command
+   7. You can pass environment variables through to the Docker container by populating a `Map` and passing it to
+      the `DockerAppsRunner.runToolCommand` method. Two environment variables are always passed:
```
[the remark this answers is not in the document]
Comment (melissachang) [reply CONTRIBUTING.md]: Done

File: src/main/java/bio/terra/cli/app/utils/AppDefaultCredentialUtils.java (hunk 1 above)
Comment (melissachang) [reply src/main/java/bio/terra/cli/app/utils/AppDefaultCredentialUtils.java:21]: Hmm, this class has 3 `ADC`, 1 `Adc`, and 1 `ApplicationDefaultCredentials`. I think `ADC` is ok.

File: src/main/java/bio/terra/cli/businessobject/User.java (hunk 2 above)
Comment (melissachang) [reply src/main/java/bio/terra/cli/businessobject/User.java:281]: I'd have to see this error in the wild, to know what a better error message would be

File: src/test/java/harness/TestUser.java (hunk 3 above)
Comment (melissachang) [reply src/test/java/harness/TestUser.java]: Done, thanks

File: CONTRIBUTING.md (hunk 4 above)
Comment (melissachang) [reply CONTRIBUTING.md]: Done, thanks

File: src/main/java/bio/terra/cli/businessobject/User.java (hunk 2 above)
Comment (jaycarlton) [reply src/main/java/bio/terra/cli/businessobject/User.java:281]: I don't know if we have user-facing docs about pet SAs anywhere, but I suppose they'll call us if there's trouble.

**Pull request merged**